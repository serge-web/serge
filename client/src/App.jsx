import { Component } from 'react'
import { Provider } from 'react-redux'
import { MuiThemeProvider as ThemeProvider } from '@material-ui/core/styles'
import theme from './themes/serge'
// import Notifications from './Components/Notifications'
// import Version from './Views/Version'
import RouterMain from './Components/Router/RouterMain'
import Store from './Store/Store'
// import '@serge/themes/App.scss'
import './themes/App.scss'


class App extends Component {
  componentDidMount () {
    document.title = 'Serge'
  }
  
  render () {
    return (
      <Provider store={Store}>
        <ThemeProvider theme={theme}>
          <RouterMain />
          {/* <Notifications /> */}
          {/* <Version /> */}
        </ThemeProvider>
      </Provider>
    )
  }
}

export default App
