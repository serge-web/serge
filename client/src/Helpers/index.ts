/***
 * This file is autogenerated, please do not edit it.
 * To generate an updated version, please run `yarn genreg`
 ***/
export { default as capitalize } from './capitalize'
export { default as checkUnique } from './check-unique'
export { default as convertMilliseconds } from './convert-milliseconds'
export { default as deepCompare } from './deep-compare'
export { default as deepCopy } from './deep-copy'
export { default as duplicateThisForce } from './duplicate-force'
export { default as configDateTimeLocal } from './config-date-time-local'
export { default as forceIcons } from './force-icons'
export { default as forceRole } from './force-role'
export { default as formatTurn } from './format-turn'
export { default as getRoleFromName } from './get-role-from-name'
export { default as getRoleFromId } from './get-role-from-id'
export { default as getSelectedItem } from './getSelectedItem'
export {
  default as handleChannelUpdates,
  handleAllInitialChannelMessages,
  handleNewMessageData,
  clipInfoMEssage
} from './handle-channel-updates'
export { default as hexToRGBA } from './hex-to-rgba'
export { default as findEmptyRolenames } from './find-empty-rolenames'
export { default as findDuplicatePasscodes } from './find-duplicate-passcodes'
export { default as getUniquePasscode } from './get-unique-passcode'
export { default as incrementGameTime } from './increment-game-time-helpers'
export { default as isArrayOfObject } from './is-array-of-object'
export { default as isBoolean } from './is-boolean'
export { default as isError } from './is-error'
export { default as isHexLight } from './is-hex-light'
export { default as isNumber } from './is-number'
export { default as isObjectEquivalent } from './is-object-equivalent'
export { default as isPlainObject } from './is-plain-object'
export { type ForceStyle } from './is-perceived-by'
export { default as isValidUrl } from './is-valid-url'
export { default as mostRecentOnly } from './most-recent-only'
export { default as newestPerRole, logTable } from './newest-per-role'
export {
  matchedForceAndRoleFilter,
  matchedAllRolesFilter,
  matchedForceFilter
} from './participant-filters'
export {
  checkLegacyParticipantStates,
  checkV3ParticipantStates,
  getParticipantStates
} from './participant-states'
// export { default as shortPoll } from './short-poll'
export { default as usePrevious } from './use-previous'
export { default as generateClassName } from './generate-class-name'
export * from './message-status-state'
export * from './message-unsent-state'
export * from './new-date'
export * from './format-date'
export { default as deleteRoleAndParts } from './delete-role-and-parts'
