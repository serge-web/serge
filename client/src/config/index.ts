/***
* This file is autogenerated, please do not edit it.
* To generate an updated version, please run `yarn genreg`
***/

export * from './globals'
export * from './routes'
export * from './enums'
export * from './action-constants'
