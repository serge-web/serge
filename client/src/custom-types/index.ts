/***
* This file is autogenerated, please do not edit it.
* To generate an updated version, please run `yarn genreg`
***/

export * from './channel-data.d'
export type { default as ChannelUI } from './channel-ui.d'
export type { default as Channel } from './channel.d'
export * from './export-items-actions.d'
export * from './export-items.d'
export type { default as ForceData } from './force-data.d'
export type { default as ForceOption } from './force-option.d'
export type { default as ForceRole } from './force-role.d'
export type { default as Force } from './force.d'
export * from './global.d'
// export { default as MapPostBack } from './map-postback-data.d'
export type { default as MarkerPerception } from './marker-perception.d'
export type { default as MessageTypes, TemplateBodysByKey } from './message-types.d'
export * from './message.d'
export type { SetCurrentViewFromURI } from './messages-action.d'
export * from './participant.d'
export * from './player-log-actions.d'
export * from './player-log.d'
export * from './player-ui-actions.d'
export type { default as PlayerUi, PlayerUiChannels, PlayerUiChatChannel } from './player-ui.d'
export type { default as WargamesState } from './WargamesState.d'
export type { default as Role } from './role.d'
export * from './root-state'
export type { default as SetWargameMessage } from './set-wargame-message.d'
export type { default as Template, TemplateBody, TemplateData } from './template.d'
export * from './turn-length.d'
export * from './wargame-action.d'
export type { default as WargameData } from './wargame-data.d'
export type { default as WargameList } from './wargame-list.d'
export type { default as WargameOverview } from './wargame-overview.d'
export type { default as Wargame } from './wargame.d'
export * from './core-mapping.d'
export * from './turn-period.d'
