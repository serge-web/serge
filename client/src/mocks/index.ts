/***
* This file is autogenerated, please do not edit it.
* To generate an updated version, please run `yarn genreg`
***/

export { default as GameMessagesMock } from './game-messages.mock'
export { default as GameMessagesMockRFI } from './game-messages-rfi.mock'
export { default as AdminMessagesMock } from './admin-messages.mock'
export { default as ChatMessagesMock } from './chat-messages.mock'
export { default as InfoMessagesMock } from './info-messages.mock'
export { default as GameChannels2 } from './game-channels-2.mock'
export { default as WargameMock } from './wargame.mock'
export { default as WargameExportedMock } from './wargame-exported.mock'
export { default as MessageTemplatesMock } from './message-templates.mock'
export * from './message-templates-by-key.mock'
export { default as forces } from './forces.mock'
export { default as atlanticForces } from './atlantic-forces.mock'
export { default as watuWargame } from './watu-wargame-mock'
export { default as preInitialisedForces } from './pre-initialised-forces.mock'
export { default as adminTabs } from './admin-tabs.mock'
export { default as feedbackMessage, feedbackMessage1, feedbackMessage2 } from './feedback-message.mock'
export { default as actionFeedbackMessages } from './action-feedback-messages.mock'
export { default as actionSetLatestFeedbackMessage } from './action-set-latest-feedback-message.mock'
export { default as actionSetLatestWargameMessage } from './action-set-latest-wargame-message.mock'
export { default as actionSetWargameMessages } from './action-set-wargame-messages.mock'
export { default as actionOpenMessage } from './action-open-message.mock'
export { default as actionCloseMessage } from './action-close-message.mock'
export * from './coa-channel-data-2.mock'
export * from './coa-messages.mock'
