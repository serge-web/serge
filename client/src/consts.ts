import uniqId from 'uniqid'
import moment from 'moment'
import ExpiredStorage from 'expired-storage'
import * as Globals from '@serge/config'
import { Wargame } from '@serge/custom-types'

/** name of the document used to store the initial wargame definition */
export const DEFAULT_SERVER = 'Nelson'
export const DEFAULT_PORT = '8080'

export const MSG_STORE = 'messages'
export const SERGE_INFO = 'serge_info'
export const CHAT_CHANNEL_ID = 'game-admin'

export const PLANNING_PHASE = 'planning'
export const ADJUDICATION_PHASE = 'adjudication'

export const ADMIN_ROUTE = '/serge/admin'
export const MESSAGE_TEMPLATE_ROUTE = '/serge/messageTemplates'
export const MESSAGE_CREATOR_BASE_ROUTE = '/serge/messageCreator'
export const CREATE_TEMPLATE_ROUTE = '/create/template'
export const EDIT_TEMPLATE_ROUTE = '/edit/template'
export const CREATE_MESSAGE_ROUTE = '/create/message'
export const EDIT_MESSAGE_ROUTE = '/edit/message'
export const GAME_SETUP_ROUTE = '/serge/gameSetup'
export const WELCOME_SCREEN_EDIT_ROUTE = '/serge/editWelcomeScreen'
export const EXPORT_ROUTE = '/serge/export'
export const EXPORT_MESSAGES_SUBROUTE = '/messages'
export const EXPORT_FORCES_SUBROUTE = '/foeces'
export const EXPORT_PRINT_SUBROUTE = '/print/:id'
export const PLAYERUI_ROUTE = '/serge/player'
export const DEMO_ROUTE = '/serge/demo'
export const STORYBOOK_ROUTE = './storybook'

export const expiredStorage = new ExpiredStorage()
export const LOCAL_STORAGE_TIMEOUT = 2592000 // one month

export const MAX_LISTENERS = 82

export const UMPIRE_FORCE = 'umpire'

export const LOCATION_PENDING = 'LocationPending' // special state where platforms can be moved at turn zero

// return states of adjudication form
export const PLAN_ACCEPTED = 'accepted'
export const PLAN_REJECTED = 'rejected'

// names of special channels
export const CHANNEL_RFI_STATUS = 'rfis'
export const CHANNEL_MAPPING = 'mapping'

// NOTE: time period to wait if server returns an error. One frequent cause of error
// during development is that the server is stopped.  We're introducing a
// throttle value to prevent the browser going into a race condition
// as it sends 1000s of requests to the server
export const ERROR_THROTTLE = 3000

// Interval for check server heartbeats
export const SERVER_PING_INTERVAL = 20000

// Nov 2019. Ian modified the server path to use the
// current URL, so we can use Heroku to provide
// review instances of the app.  In these
// review instances, we can't predict the URL, so
// were failing CORS test
export const baseUrl = () => {
  const { hostname, protocol } = window.location
  // @ts-ignore
  const host = (new URL(window.location)).searchParams.get('host')

  // NOTE: for all non-heroku deployments, we need to append the port number
  const portDetails = hostname.toLowerCase().indexOf('herokuapp') !== -1 ? '' : ':' + DEFAULT_PORT

  const res = host || `${protocol}//${hostname}` + portDetails

  return res
}

export const serverPath = (
  // @ts-ignore
  window.G_CONFIG.REACT_APP_SERVER_PATH || process.env.REACT_APP_SERVER_PATH || baseUrl() + '/'
).replace(/\/?$/, '/')

export const databasePath = `${serverPath}db/`

export const headers = {
  'Content-Type': 'application/json',
  'Access-Control-Allow-Origin': '*',
  'Access-Control-Allow-Methods': 'POST',
  'Access-Control-Allow-Headers': 'Authorization, Lang'
}

export const defaultGameInfo = {
  imageUrlSet: false,
  imageUrl: '/default_img/sergeDefault.png',
  title: 'Serge',
  description: `You have arrived at the Development Centre Gaming Facility.\n
  You will use this web-based application to interact with players from other forces, together with the umpires in the White Cell.\n
  At any point during your time here you can submit insights via the Insights button at the top-right of the gaming page.\n
  These insights could relate to the current doctrine being explored, the performance of your force, or how the game is being organised / facilitated.\n
  Thanks in advance for your participation.\n
  Maj Duncan Dare, PO1 Gaming`,
  showAccessCodes: true
}

export const forceTemplate = {
  name: '',
  uniqid: '',
  overview: 'An overview written here..',
  roles: [
    {
      name: 'CO',
      roleId: `p${uniqId.time()}`,
      isGameControl: false,
      isObserver: false,
      isInsightViewer: false
    }
  ],
  iconURL: serverPath + 'default_img/forceDefault.png',
  color: '#3dd0ff',
  umpire: false,
  dirty: false
}

export const umpireForceTemplate = {
  name: 'White',
  uniqid: 'umpire',
  overview: 'Umpire force.',
  roles: [
    {
      name: 'Game Control',
      roleId: `p${uniqId.time()}`,
      isGameControl: true,
      isObserver: true,
      isInsightViewer: true
    }
  ],
  iconURL: serverPath + 'default_img/umpireDefault.png',
  color: '#FCFBEE',
  umpire: true,
  dirty: false
}

export const channelTemplate = {
  name: '',
  uniqid: '',
  participants: []
}

export const dbDefaultPlaylogSettings = {
  wargame: 'missing',
  role: 'missing',
  activityType: {
    aType: 'pending'
  },
  activityTime: '',
  _id: new Date().toISOString(), 
  isOpen: false
}

export const dbDefaultSettings: Wargame = {
  _id: Globals.wargameSettings,
  wargameTitle: '',
  name: '',
  data: {
    overview: {
      name: 'Overview - settings',
      gameDescription: '',
      // spatialRepresentation: '',
      gameTurnTime: {
        unit: 'millis',
        millis: 43200000
      },
      realtimeTurnTime: 300000,
      timeWarning: 60000,
      // turnStrategy: '',
      gameDate: moment(new Date(), moment.ISO_8601).format(),
      showAccessCodes: true,
      logPlayerActivity: true,
      dirty: false
    },
    forces: {
      name: 'Forces',
      forces: [umpireForceTemplate],
      selectedForce: umpireForceTemplate,
      dirty: false
    },
    channels: {
      name: 'Channels',
      channels: [],
      selectedChannel: '',
      dirty: false
    }
  },
  wargameList: [],
  wargameInitiated: false,
  gameTurn: 0,
  phase: ADJUDICATION_PHASE,
  adjudicationStartTime: moment(new Date(), moment.ISO_8601).format()
}

export const FLEX_LAYOUT_MODEL_DEFAULT: any = {
  global: {
    tabSetTabStripHeight: 45,
    tabEnableClose: false,
    tabEnableRenderOnDemand: false
  },
  borders: [],
  layout: {
    type: 'row' as string,
    weight: 100,
    children: [
    ]
  }
}
