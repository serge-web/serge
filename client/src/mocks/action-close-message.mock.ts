import { CLOSE_MESSAGE } from 'src/config'
import { CloseMessageAction } from 'src/custom-types'
import { GameMessagesMock } from '.'

const actionCloseMessage: CloseMessageAction = {
  type: CLOSE_MESSAGE,
  payload: {
    "channel": GameMessagesMock[0].details.channel,
    "message": GameMessagesMock[0]
  }
}

export default actionCloseMessage
