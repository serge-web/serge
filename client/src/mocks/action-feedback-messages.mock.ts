import { SET_FEEDBACK_MESSAGES } from 'src/config'
import { SetWargameFeedbackAction } from 'src/custom-types'
import { feedbackMessage1, feedbackMessage2 } from './feedback-message.mock'

const actionFeedbackMessages: SetWargameFeedbackAction = {
  type: SET_FEEDBACK_MESSAGES,
  payload: [feedbackMessage1, feedbackMessage2]
}

export default actionFeedbackMessages
