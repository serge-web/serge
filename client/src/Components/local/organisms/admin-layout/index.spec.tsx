/* global it expect */

import React from 'react'
import renderer from 'react-test-renderer'

import AdminLayout from './index'
import { WargameMock as wargame } from 'src/mocks'

it('AdminLayout renders correctly', () => {
  const tree = renderer
    .create(<AdminLayout wargame={wargame}>Content</AdminLayout>)
    .toJSON()
  expect(tree).toMatchSnapshot()
})
