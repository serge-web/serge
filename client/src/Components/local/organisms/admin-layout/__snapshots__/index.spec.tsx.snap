// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AdminLayout renders correctly 1`] = `
<div
  className="main"
>
  <div
    className="left-panel"
  >
    <a
      className="link main"
      href="#"
      onClick={[Function]}
    >
      <svg
        className="svg-inline--fa fa-arrow-left"
      />
    </a>
  </div>
  <div
    className="container"
  >
    <div
      className="header"
    >
      <div
        className="main"
      >
        <div
          className="input-box item"
        >
          <section
            className="input-container disable-offset underline"
          >
            <div
              className="MuiFormControl-root MuiTextField-root"
            >
              <div
                className="MuiInputBase-root MuiInput-root makeStyles-root-4 makeStyles-root-6 MuiInput-underline makeStyles-underline-5 makeStyles-underline-7 MuiInputBase-formControl MuiInput-formControl"
                onClick={[Function]}
              >
                <input
                  aria-invalid={false}
                  autoFocus={false}
                  className="MuiInputBase-input MuiInput-input"
                  disabled={false}
                  name=""
                  onAnimationStart={[Function]}
                  onBlur={[Function]}
                  onChange={[Function]}
                  onFocus={[Function]}
                  required={false}
                  type="text"
                  value="Task-Group"
                />
              </div>
            </div>
          </section>
        </div>
      </div>
    </div>
    <div
      className="content"
    >
      Content
    </div>
  </div>
</div>
`;
